core/cluster/process.go

```diff
@@ -0,0 +1,38 @@
+package cluster
+
+import "time"
+
+type (
+
+	SingleProcess interface {
+		//Do runs the given function if there is no other function with that key running
+		Do(key string, fn func()) (bool, error)
+	}
+
+	DistributedLock interface {
+		//GetLock - call to requests a distributed lock - if not possible. returns "false" in case Lock is not available without blocking
+		GetLock(key string, max *time.Duration) (bool, error)
```

Comment:
Use `context.Context` instead of `time.Duration` to react on canceled contexts. Timeout can be achieved by using `context.WithTimeout` or `context.WithDeadline`.

Reply:
Rename to `Lock`.

Reply:
Lock blocks
this function should nit block
maybe there is another better name..